pkg/server/gitproviders/gitprovider.go

```diff
@@ -68,6 +76,7 @@ func (s *GitProviderService) GetConfigForUrl(repoUrl string) (*gitprovider.GitPr
 		if canHandle {
```

Comment:
Cloning a private repo of the second github account fails with this setup. This is what we need instead:

```
		if canHandle {
			_, err = gitProvider.GetRepositoryContext(gitprovider.GetRepositoryContext{
				Url: repoUrl,
			})
			if err == nil {
				return p, nil
			}
		}
```